src/datadoc/backend/model_validation.py

```diff
@@ -0,0 +1,175 @@
+
+        Mode: This validator runs after other validation.
+
+        Returns:
+            Self: The instance of the model after validation.
+        """
+        timestamp: datetime = get_timestamp_now()  # --check-untyped-defs
+        if self.dataset is not None and self.dataset.metadata_created_date is None:
+            self.dataset.metadata_created_date = timestamp
+        return self
+
+    @model_validator(mode="after")
+    def check_inherit_values(self) -> Self:
+        """Inherit values from dataset to variables if not set.
+
+        Sets values for 'data source', 'temporality type', 'contains data from', and 'contains data until' if they are None.
+
+        Mode: This validator runs after other validation.
+
+        Returns:
+            Self: The instance of the model after validation.
+        """
+        if self.variables and self.dataset is not None:
+            set_variables_inherit_from_dataset(self.dataset, self.variables)
+        return self
+
+    @model_validator(mode="after")
+    def check_obligatory_dataset_metadata(self) -> Self:
+        """Check obligatory dataset fields and issue a warning if any are missing.
+
+        Mode:
+            This validator runs after other validation.
+
+        Returns:
+            Self: The instance of the model after validation.
+
+        Raises:
+            ObligatoryDatasetWarning: If not all obligatory dataset metadata fields are filled in.
+        """
+        if (
+            self.dataset is not None
+            and num_obligatory_dataset_fields_completed(
+                self.dataset,
+            )
+            != NUM_OBLIGATORY_DATASET_FIELDS
+        ):
+            warnings.warn(
+                f"All obligatory metadata is not filled in for dataset {get_missing_obligatory_dataset_fields(self.dataset)}",
+                ObligatoryDatasetWarning,
+                stacklevel=2,
+            )
+
+        return self
+
+    @model_validator(mode="after")
+    def check_obligatory_variables_metadata(self) -> Self:
+        """Check obligatory variable fields and issue a warning if any are missing.
+
+        Mode:
+            This validator runs after other validation.
+
+        Returns:
+            Self: The instance of the model after validation.
+
+        Raises:
+            ObligatoryVariableWarning: If not all obligatory variable metadata fields are filled in.
+        """
+        if (
+            self.variables is not None
+            and num_obligatory_variables_fields_completed(
+                self.variables,
+            )
+            != NUM_OBLIGATORY_VARIABLES_FIELDS
+        ):
+            warnings.warn(
+                f"All obligatory metadata is not filled in for variables {get_missing_obligatory_variables_fields(self.variables)}",
+                ObligatoryVariableWarning,
+                stacklevel=2,
+            )
+
+        return self
+
+
+class ValidateDataset(model.Dataset):
```

Comment:
Is this needed?

Reply:
No, this is history now (but I testet field_validator)